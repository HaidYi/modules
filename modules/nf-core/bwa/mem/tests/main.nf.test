nextflow_process {

    name "Test Process BWA_MEM"
    tag "modules_nfcore"
    tag "modules"
    tag "bwa"
    tag "bwa/mem"
    tag "bwa/index"
    script "../main.nf"
    process "BWA_MEM"

    setup {
        run("BWA_INDEX") {
            script "../../index/main.nf"
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }
    }

    test("human Paired-End - with alt") {

        setup {
            run("BWA_INDEX", alias: "BWA_INDEX_HUMAN") {
                script "../../index/main.nf"
                process {
                    """
                    input[0] = [
                        [id: 'test'],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists: true)
                    ]
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = [
                    [ id:'test'],
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ]
                input[1] = BWA_INDEX_HUMAN.out.index
                input[2] = [[id: 'test'],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists: true)]
                input[3] = [[id: 'test'],file('/workspace/modules/chr21.alt', checkIfExists: true)]
                input[4] = false
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.cram,
                    process.out.csi,
                    process.out.crai,
                    process.out.versions,
                    bam(process.out.bam[0][1]).getHeader(),
                    bam(process.out.bam[0][1]).getStatistics(),
                    bam(process.out.bam[0][1]).getHeaderMD5(),
                    bam(process.out.bam[0][1]).getReadsMD5()
                    ).match()
                }
            )
        }

    }

}
