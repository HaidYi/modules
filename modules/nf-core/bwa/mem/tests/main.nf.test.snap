{
    "human Paired-End - with alt": {
        "content": [
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                "versions.yml:md5,c60680eba0f00e791c0d5a0a6e9d665f"
            ],
            [
                "@HD\tVN:1.6\tSO:unsorted\tGO:query",
                "@SQ\tSN:chr21\tLN:46709983",
                "@PG\tID:bwa\tPN:bwa\tVN:0.7.18-r1243-dirty\tCL:bwa mem -t 2 ./index/genome test_1.fastq.gz test_2.fastq.gz",
                "@PG\tID:samtools\tPN:samtools\tPP:bwa\tVN:1.21\tCL:samtools view --threads 2 -o test.bam -"
            ],
            {
                "maxReadLength": 108,
                "minReadLength": 90,
                "meanReadLength": 100,
                "maxQuality": 60,
                "minQuality": 0,
                "meanQuality": 56,
                "readCount": 533472,
                "duplicateReadCount": 0
            },
            "bd8a96bfba6499c2b82a297fbc56a80f",
            "56f239265c171f74c531dbb701f95f7e"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-02T09:46:10.2951335"
    }
}